pylint/checkers/typecheck.py

```diff
@@ -583,7 +583,7 @@ def _enum_has_attribute(
             (c.value for c in dunder_new.get_children() if isinstance(c, nodes.Return)),
             None,
         )
-        if returned_obj_name is not None:
+        if isinstance(returned_obj_name, nodes.Name):
```

Comment:
Wouldn't it be better if we actually infer the value here? Instead of handling only `nodes.Name`?

Reply:
Sorry Daniel, I'm trying to understand what you mean, but I'm struggling to figure out how we'd infer the value?

Reply:
With the function `safe_infer` (imported from `utils`) you can try to let `astroid` (an upgrade to the stdlib `ast` library) try to "infer" what the actual value of a node is.
So for ``x = 1`` `astroid` will be able to determine that `x` is a `nodes.Const` with `value = 1` instead of simply a `nodes.Name`. That way you get much more information about a particular node.

In fact, you can even infer a `nodes.Call` and try to determine what a call to a certain function actually returns, which seems to be what you're trying to do here. I should have caught this when I reviewed https://github.com/PyCQA/pylint/pull/5847/.

Looking at the code we should be able to infer what `dunder_new` returns and then do the subsequent checks for the message. You can look for `safe_infer` in the code base to see how other checkers user inference to know more about nodes.

Reply:
Ok, so I've given it a go and it doesn't seem very promising.
For all the test cases in `tests/functional/e/enum_self_defined_member_5138.py`, the result of `safe_infer` comes out as `Uninferable`.
For the new tests cases in `tests/functional/e/enum_self_defined_member_6805.py`, the result of `safe_infer` is `None`

Reply:
Interesting suggestion, but it looks like there are changes in `astroid` needed so definitely not anything to look into in the context of a backportable pylint crash fix.

I'll open a followup issue in astroid, and then maybe we can revisit this later.